### PR TITLE
replace isosurface visual method to display

### PR DESCRIPTION
**Pull request by PennyQ**
This PR replace our Isosurface viewer from using `IsoSurface` visual as shown:
image

to use `Volume` visual (rendering method='iso') and gaussian smooth from the [script](https://gist.github.com/astrofrog/c097377f665967289caf) 
image

@astrofrog Do you know why the transparency doesn't work now? As the outer Isosurface has alpha=0.5 but it's shown as opaque here?

### CODE REVIEW

File: glue_vispy_viewers/isosurface/layer_style_widget.py [hunk 1]
```diff
@@ -21,6 +21,8 @@ class IsosurfaceLayerStyleWidget(QtGui.QWidget):
     # GUI elements
     attribute = CurrentComboProperty('ui.combo_attribute')
     level = FloatLineProperty('ui.value_level')
+    # We don't have IntLineProperty?
+    step_num = FloatLineProperty('ui.step_num')
```
Comment (PennyQ) [inline glue_vispy_viewers/isosurface/layer_style_widget.py]: The UI crashed after I added this one line of code, in order to have an input textbox for setting the isosurface layer numbers. @astrofrog Do you know the problem here? It seems the ui element is not found as shown below:

> Traceback (most recent call last):
>   File "/Users/penny/Works/Gluedev/glue/glue/glue/core/qt/layer_artist_model.py", line 207, in rowsInserted
>     parent.on_artist_add(self.model().artists)
>   File "/Users/penny/Works/Gluedev/glue/glue/glue/core/qt/layer_artist_model.py", line 313, in on_artist_add
>     self.layout_style_widgets[layer_artist] = self.layer_style_widget_cls(layer_artist)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/glue_vispy_viewers-0.5.dev0-py2.7.egg/glue_vispy_viewers/isosurface/layer_style_widget.py", line 33, in **init**
>     directory=os.path.dirname(**file**))
>   File "/Users/penny/Works/Gluedev/glue/glue/glue/utils/qt/helpers.py", line 118, in load_ui
>     return load_ui(full_path, parent, custom_widgets=CUSTOM_QWIDGETS)
>   File "/Users/penny/Works/Gluedev/glue/glue/glue/external/qt.py", line 281, in load_ui
>     return _load_ui_pyqt4(path, parent)
>   File "/Users/penny/Works/Gluedev/glue/glue/glue/external/qt.py", line 293, in _load_ui_pyqt4
>     return loadUi(path, parent)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/__init__.py", line 236, in loadUi
>     return DynamicUILoader(package).loadUi(uifile, baseinstance, resource_suffix)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/Loader/loader.py", line 71, in loadUi
>     return self.parse(filename, resource_suffix, basedir)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 982, in parse
>     actor(elem)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 803, in createUserInterface
>     self.traverseWidgetTree(elem)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 781, in traverseWidgetTree
>     handler(self, child)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 478, in createLayout
>     self.traverseWidgetTree(elem)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 781, in traverseWidgetTree
>     handler(self, child)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 521, in handleItem
>     self.traverseWidgetTree(elem)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 781, in traverseWidgetTree
>     handler(self, child)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 478, in createLayout
>     self.traverseWidgetTree(elem)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 781, in traverseWidgetTree
>     handler(self, child)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 521, in handleItem
>     self.traverseWidgetTree(elem)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 781, in traverseWidgetTree
>     handler(self, child)
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 256, in createWidget
>     self.stack.push(self.setupObject(widget_class, parent, elem))
>   File "/Users/penny/anaconda/lib/python2.7/site-packages/PyQt4/uic/uiparser.py", line 226, in setupObject
>     setattr(self.toplevelWidget, name, obj)
>   File "/Users/penny/Works/Gluedev/glue/glue/glue/utils/qt/widget_properties.py", line 72, in **set**
>     widget = reduce(getattr, [instance] + self._att)
> AttributeError: 'IsosurfaceLayerStyleWidget' object has no attribute 'ui'

**Comment by PennyQ** [comment]
TODO for this PR: 
- [x] Add step number for controlling isosurface levels
- [x] Link display setting on left panel with visual

**Comment by PennyQ** [comment]
So I added settings of low and high limits, also a slider to control step number.

Next step is to enable color selection and also solve the transparency problem, so in principle we should assign color & alpha for each isosurface layer, similar to our current subset layer architecture, but it might be a lot of work. 

So a simple alternative would be using a series of color & alpha scheme, that changes with change of isosurface layer steps and will not cause occlude problem.

@astrofrog Do you have any ideas on it?

**Comment by PennyQ** [comment]
Use an individual `MultiIsoVisual` for plotting isosurface visual with multiple layers, with individual scripts here https://github.com/PennyQ/vispy_scripts/tree/master/uniform_multi_iso


- [ ] Find a solution to add a uniform variable to shader code instead of rewriting the whole Frag_shader of `MultiIsoVisual`
- [ ] Figure out the relation between limits and IsoSurface threshold, or simply clip the volume cube data with the limits before rendering
- [x] Find a smarter colormap which could accept one variable (isosurface layer num in our case) and has better visual performance

**Comment by PennyQ** [comment]
@astrofrog This viewer went pretty well with colormaps and layer number settings, the only thing that get stuck here, is setting clim using VolumeVisual `set_data` function will result isosurface with only one layer. Could you help take a look at it? Many thanks!

**Comment by PennyQ** [comment]
This should fix the test error, can we merge it first? @astrofrog

**Comment by astrofrog** [comment]
@PennyQ - there are some flake8 errors here:

https://travis-ci.org/glue-viz/glue-vispy-viewers/jobs/212204692

Can you fix these? You can run this test command locally with:

```
flake8 --max-line-length=100 --exclude=extern glue_vispy_viewers
```

(you will need flake8 installed: ``pip install flake8``)

**Review by astrofrog — APPROVED** [review]
Thanks, looks good! I removed the change to setup.py so that this is still not available by default as we need to test it a bit more first.

**Pull request merged**